File: 3423-maximum-difference-between-adjacent-elements-in-a-circular-array/3423-maximum-difference-between-adjacent-elements-in-a-circular-array.py
class Solution:
    def maxAdjacentDistance(self, nums: List[int]) -> int:
        res=abs(nums[0]-nums[-1])
        for i in range(len(nums)-1):
            if abs(nums[i]-nums[i+1])>res:
                res=abs(nums[i]-nums[i+1])
        return res
        

File: 3423-maximum-difference-between-adjacent-elements-in-a-circular-array/3423-maximum-difference-between-adjacent-elements-in-a-circular-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAdjacentDistance = function(nums) {
    let n=nums.length
    let res=Math.abs(nums[0]-nums[n-1])
    for(let i=0;i<nums.length;i++){
        if(Math.abs(nums[i]-nums[i+1])>res){
            res=Math.abs(nums[i]-nums[i+1])
        }
    }
    return res
};
console.log(maxAdjacentDistance([1,2,4]));
